dali/operators/generic/slice/slice_attr.h

```diff
@@ -97,21 +132,86 @@ class SliceAttr {
+
+        for (size_t i = 0; i < axes.size(); i++) {
+          auto dim = axes[i];
 
+          double anchor_val = 0;
+          if (start_.IsDefined()) {
+            anchor_val = start_[data_idx].data[i];
+          } else if (rel_start_.IsDefined()) {
+            anchor_val = rel_start_[data_idx].data[i] * shape[dim];
+          }
+          DALI_ENFORCE(anchor_val >= i64min && anchor_val <= i64max,
+                       make_string("anchor value out of range [", i64min, ", ", i64max,
+                                   "]. Got: ", anchor_val));
+
+          double end_val = shape[dim];
+          if (end_.IsDefined()) {
+            end_val = end_[data_idx].data[i];
+          } else if (rel_end_.IsDefined()) {
+            end_val = rel_end_[data_idx].data[i] * shape[dim];
+          } else if (shape_.IsDefined()) {
+            double shape_val = shape_[data_idx].data[i];
+            DALI_ENFORCE(shape_val >= 0 && shape_val <= i64max,
+              make_string("shape value out of range [", 0, ", ", i64max, "]. Got: ", shape_val));
+
+            end_val = anchor_val + shape_val;
+          } else if (rel_start_.IsDefined() && rel_shape_.IsDefined()) {
+            // special case - minimize the floating point error by multiplying only once after sum
+            double rel_start_val = rel_start_[data_idx].data[i];
+            double rel_shape_val = rel_shape_[data_idx].data[i];
+            DALI_ENFORCE(rel_shape_val >= 0,
+              make_string("negative shapes are not allowed. Got: ", rel_shape_val));
+
+            end_val = (rel_start_val + rel_shape_val) * shape[dim];
+          } else if (rel_shape_.IsDefined()) {
+            double shape_val = rel_shape_[data_idx].data[i] * shape[dim];
+            DALI_ENFORCE(shape_val >= 0 && shape_val <= i64max,
+                         make_string("shape value out of range [", 0, ", ", i64max,
+                                     "]. Got: ", shape_val));
+            end_val = anchor_val + shape_val;
+          }
+          DALI_ENFORCE(end_val >= i64min && i64max <= i64max,
+                       make_string("end coordinates out of range [", i64min, ", ", i64max,
+                                   "]. Got: ", end_val));
+
+          DALI_ENFORCE(end_val >= anchor_val,
+                       make_string("end coordinates can't be before start coordinates. Got: start=",
+                                   anchor_val, " end=", end_val));
+
+          slice.anchor[dim] = std::llround(anchor_val);
+          slice.shape[dim] = std::llround(end_val) - slice.anchor[dim];
+        }
+        return slice;
+      };
+  }
+
+  template <typename AnchorT, typename ShapeT, typename EndT>
+  void ProcessPositionalInputArgs(int data_idx,
+                                  const AnchorT *slice_anchor_data,
+                                  const ShapeT *slice_shape_data,
+                                  const EndT *slice_end_data) {
```

Comment:
It raised a warning in clang in the lambda - but slice_end_data is not used in this function at all. Do we need this argument at all here?